tests/lib/eslint/eslint.js

```diff
@@ -1057,6 +1057,61 @@ describe("ESLint", () => {
             await assert.rejects(async () => await eslint.lintFiles(["lib/cli.js"]), /Expected object with parse\(\) or parseForESLint\(\) method/u);
         });
 
+        // https://github.com/eslint/eslint/issues/18407
+        it("should work in case when `fs.readFile()` returns an object that is not an instance of Promise from this realm", async () => {
```

Comment:
Maybe `fs.promises.readFile()` or `fsp.readFile()`?

Reply:
Updated to `fsp.readFile()` in https://github.com/eslint/eslint/pull/18416/commits/0b8b96b3a4d2388ee103bb9e589380b4425892e0.